app/library/postgres.py

```diff
@@ -234,12 +234,31 @@ async def duplicate_scenario(scenario_id: str, session):
 
 
 async def duplicate_scene(scene_id: str, session):
-    # TODO: actual duplication in postgres
-    sample_response = {
-        "message": "Duplicated scene with id " + scene_id,
-        "id": scene_id + "3",
-    }
-    return sample_response
+    # Get scene to duplicate
+    curr_scene = get_scene(scene_id, session).as_dict()
+    object_ids = curr_scene["object_ids"]
+    try:
+        del curr_scene["id"]
+        del curr_scene["object_ids"]
+    except KeyError:
+        pass
+
+    # Create new duplicated scene without the objects
+    new_scene = await post_scene_to_postgres(curr_scene, session)
+    new_scene_id = new_scene["id"]
+
+    # Duplicate the objects and add into the scene
+    for object_id in object_ids:
+        curr_object = get_object(object_id, session).as_dict()
+        try:
+            del curr_object["id"]
+        except KeyError:
+            pass
+        await post_object_to_postgres(new_scene_id, curr_object, session)
```

Comment:
Ah, this method is going to have O(n) calls to update the scene, because on each object being created, it will add to the scene (line 268). Since in this case, we are duplicating the scene, instead of making O(n) calls, let's make O(1) call.
So, either add a parameter to  `post_object_to_postgres` like `post_object_to_postgres(scene_id: str, data: dict, session, add_to_scene=True)` and then add an if statement for the parts related to the scene in that method, OR copy the relevant parts of that method to this method, and just do one `put_scene` with all the new object IDs appended to the scene.

Reply:
good point